enter/state.c

```diff
@@ -45,11 +49,33 @@ void mutt_enter_state_free(struct EnterState **ptr)
   FREE(ptr);
 }
 
+/**
+ * mutt_enter_state_resize - Make the buffer bigger
+ * @param es  State of the Enter buffer
+ * @param num Number of wide characters
+ */
+void mutt_enter_state_resize(struct EnterState *es, size_t num)
+{
+  if (!es)
+    return;
+
+  if (num <= es->wbuflen)
+    return;
+
+  num = ROUND_UP(num, 128);
+  es->wbuflen = num;
+  mutt_mem_realloc(&es->wbuf, num * sizeof(wchar_t));
```

Comment:
Would you consider using an ARRAY for wbuf?

Reply:
Yes, I'll look into that